coding-the-legislation/40_legislation_evolutions.md

```diff
@@ -60,67 +61,71 @@ we get the output:
 
 [Read more about how to code parameters](./legislation_parameters.md#parameters-and-time).
 
-## Variable defined only for a specific time intervale
+## Variable defined until a specific date
 
-As the legislation evolves, some fiscal or benefit mechanism appear and disapear. When creating a variable, you can specify the attribute `start_date` and `stop_date` to defines in which time intervale this variable make sense.
+As the legislation evolves, some fiscal or benefit mechanism appear and disapear.  
+For every variable, you can specify the attribute `end`; it defines until when this variable makes sense.
 
-When called outside of its definition intervale, a variable will **not** execute its formula and instead **return its default value**.
+If called outside of its definition time, a variable will **not** execute its formula and instead **return its default value**.
 
 For instance:
 ```py
 class flat_tax_on_salary(Variable):
     column = FloatCol
     entity = Person
     label = u"Individualized and monthly paid tax on salaries"
-    start_date = date(2014, 01, 01)
+    end = '2014-12-31'
     definition_period = MONTH
 
-    def function(person, period, legislation):
-        ...
-
-class progressive_income_tax(Variable):
-    column = FloatCol
-    entity = Person
-    label = u"Former tax replaced by the flat tax on the 1st of Jan 2014"
-    stop_date = date(2013, 12, 31)
-    definition_period = MONTH
-
-    def function(person, period, legislation):
+    def formula(person, period, legislation):
         ...
 ```
 
+will return a value calculated by its `formula` function if it is called for any date before its `end = '2014-12-31'` date. 
+While it will return the default OpenFisca value for `FloatCol` if it is called after its `end` date.
+
 Note that:
-- A variable can of course have both a `start_date` and a `stop_date`.
-- The `stop_date` is the last day a formula is valid, and not the first day it is not valid anymore.
+- The `end` is the last day a variable and its formulas are valid (not the first day of unvalidity).
+- The `end` value is a string of 'YYYY-MM-DD' format where YYYY, MM and DD are respectively a year, month and day.
 - When defining a date, the month is given **before** the day.
 
 ## Formula evolving over time
 
 Some fiscal or benefit mechanism significantly evolve over time, with bigger changes than a simple parameter adjustement.
 
-For instance, let's assume that from the 1st of Jan. 2017, our previous `flat_tax_on_salary` will not apply on the first `1000` earned by the person. We can implement this rule with a `DatedVariable`:
+For instance, let's assume that from the 1st of Jan. 2017, our previous `flat_tax_on_salary` will not apply on the first `1000` earned by the person. 
+We can implement this rule with dated formulas:
```

Comment:
“with dated formulas” → “by dating the formulas” (make it sound more like an action on a concept rather than introducing a new concept)